### PR TITLE
Add PRIM_IS_POD_TYPE and PRIM_IS_EXTERN_TYPE to dyno

### PR DESCRIPTION
**Pull request by dlongnecke-cray**
Add implementations of two primitives to the new compiler.

Reviewed by @mppf. Thanks!

### CODE REVIEW

File: frontend/lib/resolution/resolution-queries.cpp [hunk 1]
```diff
@@ -3837,23 +3837,51 @@ const TypedFnSignature* tryResolveInitEq(Context* context,
...
+}
+
+static const TypedFnSignature*
+tryResolveAssignHelper(Context* context,
+                       const uast::AstNode* astForScopeOrErr,
+                       const types::Type* lhsType,
+                       const types::Type* rhsType,
+                       bool asMethod) {
+  auto qtLhs = QualifiedType(QualifiedType::CONST_REF, lhsType);
+  auto qtRhs = QualifiedType(QualifiedType::CONST_REF, rhsType);
```
Comment (mppf) [inline frontend/lib/resolution/resolution-queries.cpp:3871]: `owned` will have an `=` that mutates the RHS. I don't htink that's relevant here, so you probably want `QualifiedType::VAR` here.

File: frontend/lib/types/Type.cpp [hunk 2]
```diff
@@ -230,6 +232,66 @@ const CompositeType* Type::getCompositeType() const {
...
+  return QUERY_END(ret);
+}
+
+bool Type::isPod(Context* context, const Type* t) {
+  if (t->isUnknownType() || t->isErroneousType() ||
+      t->isAnyType()) return false;
+  if (t->hasPragma(context, uast::PRAGMA_IGNORE_NOINIT)) return false;
+  if (t->hasPragma(context, uast::PRAGMA_ATOMIC_TYPE)) return false;
+  if (t->hasPragma(context, uast::PRAGMA_SYNC)) return false;
+  if (t->hasPragma(context, uast::PRAGMA_SINGLE)) return false;
```
Comment (mppf) [inline frontend/lib/types/Type.cpp:282]: there is also `pragma "plain old data"` to check for here. It's used on `record _range` today.

File: frontend/test/resolution/testTypePropertyPrimitives.cpp [hunk 3]
```diff
@@ -563,6 +567,92 @@ static void test11() {
...
+            }
+            record r6 {
+              proc init=(rhs: r6) {}
+            }
+            record r7 {
+              operator=(lhs: r7, rhs: r7) {}
+            }
+            record r8 {}
+            operator=(lhs: r8, rhs: r8) {}
+            class c1 { var x: int; }
```
Comment (mppf) [inline frontend/test/resolution/testTypePropertyPrimitives.cpp:591]: Can you add to the test a record with compiler-generated `=` etc but that contains something non-POD?

**Review by mppf — APPROVED** [review]

File: frontend/test/resolution/testTypePropertyPrimitives.cpp (hunk 3 above)
Comment (dlongnecke-cray) [reply frontend/test/resolution/testTypePropertyPrimitives.cpp:591]: I have record `r3` which contains an `integral` field. Does that work?

Comment (mppf) [reply frontend/test/resolution/testTypePropertyPrimitives.cpp:591]: IMO the behavior around `isPOD(someGenericType)` is weird & I'd be looking for a case that doesn't fall into that.

Comment (dlongnecke-cray) [reply frontend/test/resolution/testTypePropertyPrimitives.cpp:591]: Ok, I've added a case for a record that contains an owned class.

**Pull request merged**